Include/internal/pycore_stackref.h

```diff
@@ -96,6 +98,7 @@ PyStackRef_IsNone(_PyStackRef ref)
 static inline PyObject *
 _PyStackRef_AsPyObjectBorrow(_PyStackRef ref, const char *filename, int linenumber)
 {
+    assert((ref.index & 1) == 0);
```

Comment:
Is this assertion checking that it's not tagged? Shouldn't it look at 2 bits?

Reply:
This is when `Py_STACKREF_DEBUG` is defined. We only use one tag bit in that case.

Reply:
It can cause confusion that there is `PyStackRef_IsTaggedInt` and `is_tagged_int` and they are different.